sw/host/ot_certs/src/cbor.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Eq, Serialize)]
+#[repr(u8)]
+pub enum MajorType {
+    Uint = 0,
+    Sint = 1,
+    Bstr = 2,
+    Tstr = 3,
+    Array = 4,
+    Map = 5,
+}
+
+pub fn arg_size(arg: u64) -> u64 {
+    if arg <= 23 {
+        0
+    } else if arg <= u8::MAX.into() {
+        1
+    } else if arg <= u16::MAX.into() {
+        2
+    } else if arg <= u32::MAX.into() {
+        4
+    } else {
+        8
+    }
+}
+
+fn header(kind: MajorType, arg: u64) -> Vec<u8> {
```

Comment:
I think I can guess what this function does but it would be nice to document it

Reply:
Done.